### PR TITLE
Post comments were invisible in Individual Post Page with Image

### LINKED ISSUE #2333: Bug: Post comments are invisible in Individual Post Page.
**Describe the bug**
`Post comments` in `individual post page` are empty and `invisible` despite the `existence` of comments.

**To Reproduce**

1. Login and Join an organisation.
2. Create a post in case of empty posts.
3. Find the created post in news feed and click on its comment option.
4. Add a new comment.
5. Navigate back to news feed.
6. Once again click on same post comment option.

**Expected behavior**
The previously created comment should be visible.

**Actual behavior**
The comment list is empty despite the comment icon showing non zero value.

**Screenshots**

https://github.com/PalisadoesFoundation/talawa/assets/120930148/981d45d6-854d-4823-9028-f3d2a058ff81

**Additional details**

**Potential internship candidates**
Please read this if you are planning to apply for a Palisadoes Foundation internship https://github.com/PalisadoesFoundation/talawa/issues/359

### PR DESCRIPTION
**Pull request by imshivam-gupta**
**What kind of change does this PR introduce?**

BugFix: Comments were not visible for posts with images. 

**Issue Number:**

Fixes #2333 

**Did you add tests for your changes?**

Yes

**Snapshots/Videos:**

https://github.com/PalisadoesFoundation/talawa/assets/93479496/2eb8603d-b3dc-48d2-9226-3d1af27aafab

first
seconf
third

**If relevant, did you update the documentation?**

**Summary**

Comments were not visible for posts with images

**Does this PR introduce a breaking change?**

No

**Other information**

The bug was there because app was trying to get the complete post data for fetching comments of the post and the `imageurl`  was to be sent in proper format by api for this. That's how `fetchPostsByOrganisation` was working. Now since only comments were needed I removed the extra data fecthed along with comments.

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa/blob/master/CONTRIBUTING.md)?**

Yes

### CODE REVIEW

**Comment by Dante291** [comment]
@imshivam-gupta 

1. I am not facing this bug 

[untitled.webm](https://github.com/PalisadoesFoundation/talawa/assets/103507835/79415d98-1391-4ade-bf7d-d9bf72269036)

2. fetching all post data can not be a problem as we are then extracting comments from the Query Result, so fetching extra data for post shouldn't even be a problem here 
 
image


Further, this Query can be used to fetch data for a whole post for future use case, suppose user have to edit a post which could be handled in a new screen where whole post data will be required, So I wouldn't recommend changing name of the query as it makes sense and will be helpful for future use case.

**Comment by imshivam-gupta** [comment]
> @imshivam-gupta
> 
> 1. I am not facing this bug
> 
>  untitled.webm 
> 2. fetching all post data cannot be a problem as we are then extracting comments from the Query Result, so fetching extra data for post shouldn't even be a problem here
> 
> image
> 
> Further, this Query can be used to fetch data for a whole post for future use case, suppose user have to edit a post which could be handled in a new screen where whole post data will be required, So I wouldn't recommend changing name of the query as it makes sense and will be helpful for future use case.

@Dante291 
1. Try to view comments for post with image.
2. I am not saying that we should remove this functionality I am just trying to fix the current issue and at this point of time there is no need of fetching individual post item in app. Still I can create a new issue for feature request but that will require api support as well so issue needs to be raised in both repositories ( In api for getPost query because currently it is not working to fetch image we have to create a new mutation).

**Comment by Dante291** [comment]
As we are currently not focusing on new features in the app and also this might not be planned, Do create an Issue to fix the mutation to fetch Image for a post(which has image in it) as that is the sole reason for occurrence of this bug but for now this fix will do the work i guess.

**Comment by Azad99-9** [comment]
@Dante291 did you check it for posts with images ? Are you not facing the issue ?

**Comment by Dante291** [comment]
> @Dante291 did you check it for posts with images ? Are you not facing the issue ?

Yeah facing this bug for posts with images and as explained by shivam this issue is from the backend side

**Comment by Azad99-9** [comment]
@imshivam-gupta it would be more contextful if you put out a video.

**Comment by Cioppolo14** [comment]
@noman2002 @Azad99-9 @Dante291 Can we discuss this on the issue? #2333 Is this a valid issue?

**Comment by Dante291** [comment]
> @noman2002 @Azad99-9 @Dante291 Can we discuss this on the issue? #2333 Is this a valid issue?

Basically issue is from backend side, getPostById mutation need to be fixed to fetch Image for a post (if it has one). That is why this bug is not occurring for posts which doesn't have an Image. This small fix by Shivam will fix the issue but what I suggest is fixing the mutation from backend side would make more sense.

**Comment by Cioppolo14** [comment]
@literalEval @noman2002 Can you review this PR?

**Comment by Azad99-9** [comment]
@imshivam-gupta please do the changes mentioned in [this](https://github.com/PalisadoesFoundation/talawa/issues/2333#issuecomment-1907678196) comment, so that you don't have to modify `getPostById` query.

**Comment by imshivam-gupta** [comment]
Yes I will try to do this by tomorrow I was not well today

File: lib/utils/post_queries.dart [hunk 1]
```diff
@@ -47,31 +47,12 @@ class PostQueries {
...
-            image
-          }
-          organization{
-            _id
-          }
-          likedBy{
-            _id
-          }
+          _id,
           comments{
```
Comment (Ayush0Chaudhary) [inline lib/utils/post_queries.dart:56]: Is there other query which fetch post by ID? or getPostById was only used for comments?

Comment (Azad99-9) [reply lib/utils/post_queries.dart:56]: @Ayush0Chaudhary please have a look at this [comment](https://github.com/PalisadoesFoundation/talawa/issues/2333#issuecomment-1907678196).

**Review by noman2002 — CHANGES_REQUESTED** [review]
Fix the failing test.

**Comment by imshivam-gupta** [comment]
> Fix the failing test.

Sure I will fix this

**Comment by Azad99-9** [comment]
@imshivam-gupta please check the overall codecov of repo has dropped to `0.03`.

**Comment by imshivam-gupta** [comment]
> @imshivam-gupta please check the overall codecov of repo has dropped by `0.03`.

Sorry, I forgot to test the feature `getPostById`. Now the question is whether to keep this query or not since it is not used in codebase for now and there is issues in using this query from the api side.

**Comment by Azad99-9** [comment]
For now write tests for the query. When we actually use it. Then we can modify it accordingly.

**Review by Azad99-9 — APPROVED** [review]
Looks good to me.

**Review by noman2002 — APPROVED** [review]
LGTM

**Review by palisadoes — APPROVED** [review]

**Pull request merged**